### DISCUSSION TITLE
Room names, strings only?

### DISCUSSION THREAD

**Discussion by Tomaster134** [miguelgrinberg/Flask-SocketIO#2041] [Q&A] [answered]
I was digging through the documentation but I couldn't find anything specific, is it possible to have room names as something other than strings? I was wanting to have the room names correlate to a grid, but it seemed like when I used tuples it wasn't properly handling the room name. Let me know if I've missed something, thanks!

**Comment by miguelgrinberg** [comment]
I have always assumed that room names would be strings, but I would imagine any type that can be a dictionary key would work. What error(s) did you get?

**Reply by Tomaster134** [reply]
I don't seem to be getting any errors, at least from Flask and Python. It says it's successfully connected to a room, and it seems to be catching emitted events from the client, but it doesn't seem to be sending anything to the room itself from the server. When I have the room name assigned as "(0,0)" it works, but if I change it to (0,0) my client side never gets anything.

[The project is on github](https://github.com/Tomaster134/myTextRPG) if you want to take a look. It's very much a WIP, but the basic chatroom functionality was working before I started fiddling with tuples.

**Reply by miguelgrinberg** [reply]
> and it seems to be catching emitted events from the client,

I'm not sure I understand this, because clients cannot emit to rooms. Clients can only emit to the server, the server then forwards the message to a room if appropriate.

But in any case, I'll check. I can't promise this will be fixed though. If it works, it is actually just as a side effect of duck typing (which is great), but I don't think it makes sense to add complicated logic to support such this odd use case.

**Reply by Tomaster134** [reply]
> > and it seems to be catching emitted events from the client,
> 
> I'm not sure I understand this, because clients cannot emit to rooms. Clients can only emit to the server, the server then forwards the message to a room if appropriate.

My apologies, that's what I meant when I said that. Server seems to get the messages from the client, but doesn't seem to be able to then forward to the room the client is supposedly in. This is a learning project for me, so it's my first time encountering a lot of the terminology.

>But in any case, I'll check. I can't promise this will be fixed though.

No worries! I should be able to convert on the backend to what I need, I just wanted to see if I was missing something in the documentation or if this use case had ever been explored before. I appreciate you taking the time to check!

**Reply by miguelgrinberg** [reply, marked as answer]
I've found what the issue is. You can use non-string objects as room names, as long as they are not list-like or tuple-like. When you pass a list or tuple, it is assume that you are passing a collection of rooms, so I'm treating each component as a room name. So using `(0, 1)` would send to rooms `0` and `1`.

**Reply by Tomaster134** [reply]
Fascinating. That makes sense, I appreciate you taking the time to figure out why it was behaving like that!
